lib/curl/curl.nit

```diff
@@ -122,71 +122,158 @@ class CurlHTTPRequest
+	# complying with RFC7231
+	#
+	# TODO:Replace with an enum when available
+	#
+	var method : nullable String
+
 	# Execute HTTP request
 	#
 	# By default, the response body is returned in an instance of `CurlResponse`.
 	# This behavior can be customized by setting a custom `delegate`.
 	fun execute: CurlResponse
 	do
+		self.curl.native = new NativeCurl.easy_init
 		if not self.curl.is_ok then return answer_failure(0, "Curl instance is not correctly initialized")
 
 		var success_response = new CurlResponseSuccess
 		var callback_receiver: CurlCallbacks = success_response
-		if self.delegate != null then callback_receiver = self.delegate.as(not null)
+		var err : CURLCode
+
+		# Prepare request
+		err = prepare_request(callback_receiver)
+		if not err.is_ok then return answer_failure(err.to_i, err.to_s)
+
+		# Perform request
+		var err_resp = perform
+		if err_resp != null then return err_resp
+
+		var st_code = self.curl.native.easy_getinfo_long(new CURLInfoLong.response_code)
+		if not st_code == null then success_response.status_code = st_code
+
+		self.curl.native.easy_clean
+
+		return success_response
+	end
+
+	protected fun prepare_request(callback_receiver : CurlCallbacks) : CURLCode
+	do
+		var err : CURLCode
+
+		# cURL options and delegates
+		err = set_curl_options
+		if not err.is_ok then return err
+
+		# Callbacks
+		err = set_curl_callback(callback_receiver)
+		if not err.is_ok then return err
+
+		# HTTP Header
+		err = set_curl_http_header
+		if not err.is_ok then return err
+
+		# Set HTTP method and body
+		err = set_method
+		if not err.is_ok then return err
+		err = set_body
+
+		return err
+	end
+
+	# Set cURL parameters according to assigned HTTP method set in method
+	# attribute and body if the method allows it according to RFC7231
+	protected fun set_method : CURLCode
+	do
+		var err : CURLCode
+
+		if self.method=="GET" then
+			err=self.curl.native.easy_setopt(new CURLOption.get, 1)
+
+		else if self.method=="POST" then
+			err=self.curl.native.easy_setopt(new CURLOption.post, 1)
+
+		else if self.method=="DELETE" then
+			err=self.curl.native.easy_setopt(new CURLOption.custom_request, "DELETE")
+
+		else if self.method=="HEAD" then
+			err=self.curl.native.easy_setopt(new CURLOption.no_body,1)
+
+		else
+			err=self.curl.native.easy_setopt(new CURLOption.custom_request,self. method.as(Object))
+		end
+		return err
+	end
+
+	# Set request's body
+	protected fun set_body : CURLCode
+	do
+		var err
+		var data = self.data
+		if data != null then
+			var postdatas = data.to_url_encoded(self.curl)
+			err = self.curl.native.easy_setopt(new CURLOption.postfields, postdatas)
+			if not err.is_ok then return err
+		else if body != null then
+			err = self.curl.native.easy_setopt(new CURLOption.postfields, body.as(not null))
+			if not err.is_ok then return err
+		end
+		return new CURLCode.ok
+	end
 
+	# Set cURL options such as delegate, follow location, URL, user agent
+	# and address family
```

Comment:
You can use a longer line as the first line of the doc, or declare a short title and continue with a longer description. The autodoc tools only use the first line as title, they don't take into account line wrapping.